alphadia/libtransform.py

```diff
@@ -602,6 +604,58 @@ def forward(self, input: SpecLibBase) -> SpecLibBase:
         return input
 
 
+class MultiplexLibrary(ProcessingStep):
+    def __init__(self, multiplex_mapping: dict) -> None:
+        """Initialize the MultiplexLibrary step."""
+
+        self._multiplex_mapping = multiplex_mapping
+
+    def validate(self, input: str) -> bool:
+        """Validate the input object. It is expected that the input is a path to a file which exists."""
+        valid = True
+        valid &= isinstance(input, SpecLibBase)
+
+        # check if all modifications are valid
+        for _, channel_multiplex_mapping in self._multiplex_mapping.items():
+            for key, value in channel_multiplex_mapping.items():
+                for mod in [key, value]:
+                    print(mod)
+                    if mod not in MOD_DF.index:
+                        logger.error(f"Modification {mod} not found in input library")
+                        valid = False
```

Comment:
we're not exiting the loops here to show all `mod`s that are not in the library as not to play "error ping pong", as Magnus says? otherwise we could save some iterations here

Reply:
cf. https://github.com/MannLabs/alphabase/pull/199#discussion_r1680593095

Reply:
Yes, exactly. The `validate` method should collect and print errors and fail afterwards